### PR TITLE
Sonar refactors

### PR DESCRIPTION
**Pull request by brobi1999**
Refactored the bugs discovered by the Sonar tool

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -9,6 +9,8 @@
 */
 public class TorpedoStore {
 
+  private Random generator = new Random();
+
```
Comment (R4zor25) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:13]: Jól hangzik

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (R4zor25) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: Elfogadva

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
...
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+
```
Comment (R4zor25) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:41]: felesleges sor kihagyás

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+    
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (R4zor25) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:46]: Fixed

**Review by R4zor25 — CHANGES_REQUESTED** [review]
Elfogadva

**Review by brobi1999 — COMMENTED** [review]
pr should be ok

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (brobi1999) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:13]: ok

**Review by brobi1999 — COMMENTED** [review]
ok

**Pull request merged**